Fix bug about enrolled contest page

RoboG-11 opened the pull request:
# Description

Clicking on the Contest enrolled link from the drop down is giving 400 error. This behavior was because the filter was participating and must be a signedup.

Fixes: #8018

# Checklist:

- [x] The code follows the [coding guidelines](https://github.com/omegaup/omegaup/wiki/Coding-guidelines) of omegaUp.
- [x] The tests were executed and all of them passed.
- [ ] If you are creating a feature, the new tests were added.
- [ ] If the change is large (> 200 lines), this PR was split into various Pull Requests. It's preferred to create one PR for changes in controllers + unit tests in PHPUnit,  and then another Pull Request for UI + tests in Jest, Cypress or both.

pabo99 reviewed (commented):
Could you please add a screencast of how it looks after fixing the bug?

RoboG-11 commented:
@pabo99 This is how looks before:
![image](https://github.com/user-attachments/assets/07af1f09-962c-421e-b632-e9fa2ed871e2)

And this looks after changes
![image](https://github.com/user-attachments/assets/bf69679c-7644-4081-ab29-b51b9309b45c)

pabo99 commented:
Excellent.

Just one more change: could you add a `Cypress` test to validate that the link is redirecting correctly?

The test should do the following:

The user logs in.
Then, they navigate to the dropdown option and select "Enrolled contests."
Wait for the new page to load.
Verify that the "Filter by" dropdown has the "I am participant" option selected.
You can check the existing tests as a reference for creating the new one.

If you have any questions, feel free to contact us for help.

RoboG-11 commented:
@pabo99 I added the necessary changes to fix the bug and also added the requested tests. Attached is a video showing the final solution.

https://github.com/user-attachments/assets/a236a2dc-a546-43c3-8035-a5daa60ade9a

pabo99 commented:
Oh no! You have a conflict!

If this is your first time resolving conflicts on GitHub, I’ve created a document to guide you through the process:

https://docs.google.com/document/d/10kSpSUAX3AIBWPeeR1WQcKGkHT9sIYi70vLWqQdp9Aw/edit?usp=sharing

If you have any questions, don’t hesitate to ask us.

pabo99 commented on cypress/e2e/navigation.cy.ts:
```diff
@@ -72,4 +72,36 @@ describe('Navigation Test', () => {
+    cy.get('[data-nav-user]').click();
+    cy.get('[data-nav-user-contests]').should('be.visible');
+
+    cy.get('[data-nav-user-contests]').first().click();
```
En las pruebas de Cypress (y en general en todas las pruebas)  evitamos el uso de selectores posicionales. 

¿A qué me refiero? Que en un dado caso que se agreguen mas elementos que coincidan con el selector y estos aparezcan antes del que se está haciendo referencia, esta prueba va a fallar.

Así que estuve revisando y `data-nav-user-contests` se está usando en dos opciones del menú que te redirigen a Concursos. 

Puedes modificar en el componente de Navbar el `data-attribute` de los `contests enrolled` por `data-nav-user-contests-enrolled` y acá puedes usar ese nuevo `data-attribute` y quitar el método `first()`:
 ```suggestion
     cy.get('[data-nav-user-contests-enrolled]').click();
```
no olvides también cambiar el selector de la línea 82

pabo99 commented on cypress/e2e/navigation.cy.ts:
```diff
@@ -72,4 +72,36 @@ describe('Navigation Test', () => {
+        ),
+    );
+
+    cy.get('button.dropdown-toggle-no-caret').eq(1).click();
```
Acá también puedes evitar el uso del selector posicional. Parece que el selector que estás usando es muy genérico. 

Así que la solución puede ser agregar `data-attributes` en los dos dropdowns, el primero puede llamarse `data-dropdown-order` y el que te interesa en esta prueba pudiera llamarse `data-dropdown-filter`.

De esta forma, acá cambiarías tu selector por algo así:
```suggestion
    cy.get('[data-dropdown-filter]>button').click();
```

pabo99 commented on cypress/e2e/navigation.cy.ts:
```diff
@@ -72,4 +72,36 @@ describe('Navigation Test', () => {
+    cy.get('[data-filter-by-signed-up].dropdown-item')
+      .should('exist')
+      .and('be.visible')
+      .click();
```
Este `click()` no debería lanzarse aquí, ya que se está seleccionando el elemnto que vamos a validar en la línea 104 y la prueba nunca va a fallar. 

Así que lo puedes quitar y si todo sale bien, la prueba debe serguir funcionando

pabo99 reviewed (commented):
Parece que con los comentarios que te dejé ya estaría bien este cambio

pabo99 commented on cypress/e2e/navigation.cy.ts:
```diff
@@ -72,4 +72,36 @@ describe('Navigation Test', () => {
+    cy.get('[data-nav-user]').click();
+    cy.get('[data-nav-user-contests-enrolled]').should('be.visible');
+
+    cy.get('[data-nav-user-contests-enrolled]').first().click();
```
Casi! Ya no necesitas indicar cual de los selectores quieres porque ya es único:
```suggestion
    cy.get('[data-nav-user-contests-enrolled]').click();
```

pabo99 commented on frontend/www/js/omegaup/components/arena/ContestListv2.vue:
```diff
@@ -448,6 +453,7 @@ export default class ArenaContestList extends Vue {
 
   beforeDestroy() {
     window.removeEventListener('scroll', this.handleScroll);
+    this.fetchInitialContests();
```
Acá para que quieres obtener los concursos iniciales cuando vas a destruir el componente?

Quizás esto se vino al arreglar los conflictos.

pabo99 reviewed (approved):
Muchas gracias por arreglar esto y por la paciencia para que sea aprobado!

RoboG-11 commented:
@pabo99 Te agradezco enormemente que siempre hayas respondido mis dudas que tuve corregía este error. Espero poder seguir haciendo más contribuciones al proyecto. Saludos!

pabo99 commented:
Oh, no!

Nuevamente hay conflictos :(

Pero creo que en esta ocasión deberían ser fáciles de solucionar. Quizás sea bueno que aceptes los cambios entrantes y luego vuelves a plicar el cambio en la línea que propones.


Cualquier duda, nos contactas!
